include/mender-deployment-data.h

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file      mender-deployment-data.h
+ * @brief     Mender Deployment Data interface
+ *
+ * Copyright Northern.tech AS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MENDER_DEPLOYMENT_DATA_H__
+#define __MENDER_DEPLOYMENT_DATA_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#include "mender-utils.h"
+#include "cJSON.h"
+
+#define MENDER_DEPLOYMENT_DATA_KEY_VERSION                "version"
+#define MENDER_DEPLOYMENT_DATA_KEY_ID                     "id"
+#define MENDER_DEPLOYMENT_DATA_KEY_ARTIFACT_NAME          "artifact_name"
+#define MENDER_DEPLOYMENT_DATA_KEY_PAYLOAD_TYPES          "payload_types"
+#define MENDER_DEPLOYMENT_DATA_KEY_PROVIDES               "provides"
+#define MENDER_DEPLOYMENT_DATA_KEY_NAME                   "name"
```

Comment:
I still don't like this name of the field. If it's a state I don't see why we should call it "name".

Reply:
It is what it is called in the C++ client. But I don't mind changing it. Should we call it `"state_name"`, or just `"state"`?

Reply:
FYI, I renamed it to `state` in my task that builds on this, but I guess both works

Reply:
Okey, then we do `state` ;)